execution/execution.go

```diff
@@ -123,7 +122,7 @@ func newCall(e *parser.Call, storage *engstore.SelectorPool, opts *query.Options
 	if e.Func.Name == "timestamp" {
 		switch arg := e.Args[0].(type) {
 		case *logicalplan.VectorSelector:
-			// We push down the timestamp function into the scanner through the hints.
+			arg.SelectTimestamp = true
```

Comment:
Looking at this again, shouldn't we set this in the logical plan?

Reply:
Yes it's a temporary hack here!